src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -177,125 +199,200 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
+       */
+      (upstream forall { _.nonEmpty }) && dependenciesResolved
+    }
+
+    def isRunnable(entry: ExecutionStoreEntry) = {
+      val (key, status) = entry
+      status == ExecutionStatus.NotStarted && arePrerequisitesDone(key)
+    }
+
+    def findRunnableEntries: Traversable[ExecutionStoreEntry] = executionStore filter isRunnable
+
+    val runnableEntries = findRunnableEntries
+
+    val runnableCalls = runnableEntries collect { case(k: CallKey, v) => k.scope }
+    if (runnableCalls.nonEmpty)
+      log.info(s"$tag starting calls: " + runnableCalls.map(_.fullyQualifiedName).toSeq.sorted.mkString(", "))
+
+    val entries: Traversable[Try[Iterable[ExecutionStoreKey]]] = runnableEntries map {
+      case (k: ScatterKey, v) => processRunnableScatter(k, v)
+      case (k: CollectorKey, v) => processRunnableCollector(k)
+      case (k: CallKey, v) => processRunnableCall(k, v)
+      case (k, v) =>
+        val message = s"$tag Unknown entry in execution store:\nKEY: $k\nVALUE:$v"
+        log.error(message)
+        Failure(new UnsupportedOperationException(message))
+    }
+
+    entries.find(_.isFailure) match {
+      case Some(failure) => failure
+      case _ => Success(entries.flatMap(_.get))
     }
   }
-  /* Return a Markdown table of all entries in the database */
-  private def symbolsMarkdownTable(): Option[String] = {
-    val header = Seq("SCOPE","NAME","I/O","TYPE","VALUE")
-    val maxColChars = 100
-    val rows = fetchAllEntries map { entry =>
-      val valueString = entry.wdlValue match {
-        case Some(value) => s"(${value.wdlType.toWdlString}) " + value.valueString
-        case _ => ""
-      }
-      Seq(
-        entry.key.scope,
-        entry.key.name,
-        if (entry.key.input) "INPUT" else "OUTPUT",
-        entry.wdlType.toWdlString,
-        if (valueString.length > maxColChars) valueString.substring(0, maxColChars) else valueString
-      )
-    } toSeq
-
-    rows match {
-      case r: Seq[Seq[String]] if r.isEmpty => None
-      case _ => Some(TerminalUtil.mdTable(rows, header))
+
+  private def lookupNamespace(name: String): Try[WdlNamespace] = {
+    workflow.namespace.namespaces find { _.importedAs.contains(name) } match {
+      case Some(x) => Success(x)
+      case _ => Failure(new WdlExpressionException(s"Could not resolve $name as a namespace"))
     }
   }
 
-  def fetchLocallyQualifiedInputs(call: Call): Future[Map[String, WdlValue]] = {
-    // TODO: This assumes that each Call has a parent that's a Workflow, but with scatter-gather that might not be the case 
-    val parentWorkflow = call.parent map { _.asInstanceOf[Workflow] } getOrElse {
-      throw new WdlExpressionException("Expecting 'call' to have a 'workflow' parent.")
+  private def lookupCall(key: ExecutionStoreKey, workflow: Workflow)(name: String): Try[WdlObject] = {
+    workflow.calls find { _.name == name } match {
+      case Some(matchedCall) =>
```

Comment:
When I'm looking at cases I want to be able to determine what the shape of all the possibilities will bring in a rapid fashion. If I had to scroll down nearly 20 lines, including identifying the proper other cases since there are multiple sub-cases in there that's hard to do. 

Sure, fine, sometimes it doesn't make sense. But here we needed several lines just to explain what the heck was going on in that blob of stuff. That implies it's complicated functionality and it'd be a lot simpler to reason about if there was a single phrase explaining what it was doing. If I care about the specifics I can drill in.